lua/weapons/weapon_tag.lua

```diff
@@ -0,0 +1,150 @@
+
+AddCSLuaFile()
+
+SWEP.PrintName = "Tag"
+SWEP.Author = "Ugleh"
+SWEP.Purpose = "Tag someone within 30 seconds or you will die."
+SWEP.Slot = 0
+SWEP.SlotPos = 4
+SWEP.Weight					= 99
+SWEP.AutoSwitchTo			= true
+SWEP.AutoSwitchFrom			= false
+
+SWEP.Spawnable = true
+
+SWEP.ViewModel = Model( "models/weapons/c_arms.mdl" )
+SWEP.WorldModel = ""
+SWEP.ViewModelFOV = 54
+SWEP.UseHands = true
+
+SWEP.Primary.ClipSize = -1
+SWEP.Primary.DefaultClip = -1
+SWEP.Primary.Automatic = true
+SWEP.Primary.Ammo = "none"
+
+SWEP.Secondary.ClipSize = -1
+SWEP.Secondary.DefaultClip = -1
+SWEP.Secondary.Automatic = true
+SWEP.Secondary.Ammo = "none"
+
+SWEP.DrawAmmo = false
+
+SWEP.HitDistance = 48
+
+
+local justReloaded = 0
+local initializeTime = 0
+local tagTimer = 35
+function SWEP:Initialize()
```

Comment:
According to the wiki, SWEP:Initialize is sometimes not called clientside, so this is unreliable. Can you instead do self:SetNWFloat("initTime", CurTime()) from the serverside?

Reply:
This is the only thing I am not wrapping my head around. Because Initialize sometimes doesn't run clientside should I just set initializeTime inside the if SERVER statement? If I use the code you provided is initTime a local variable for that class?